docs/sphinx/source/contributing.rst

```diff
@@ -73,17 +79,108 @@ environment.
 
 You must include documentation and unit tests for any new or improved
 code. We can provide help and advice on this after you start the pull
-request.
+request. See the Testing section below.
+
+
+.. _pull-request-scope:
+
+Pull request scope
+------------------
+
+This section can be summed up as "less is more".
+
+A pull request can quickly become unmanageable if too many lines are
+added or changed. "Too many" is hard to define, but as a rule of thumb,
+we encourage contributions that contain less than 50 lines of primary code.
+50 lines of primary code will typically need at least 250 lines
+of documentation and testing. This is about the limit of what the
+maintainers can review on a regular basis.
+
+A pull request can also quickly become unmanageable if it proposes
+changes to the API in order to implement another feature. Consider
+clearly and concisely documenting all proposed API changes before
+implementing any code. Modifying
+`api.rst <https://github.com/pvlib/pvlib-python/blob/master/docs/sphinx/source/api.rst>`_
+and/or the latest `whatsnew file <https://github.com/pvlib/pvlib-python/tree/master/docs/sphinx/source/whatsnew>`_
+can help formalize this process.
+
+Of course, sometimes it is necessary to make a large pull request. We
+only ask that you take a few minutes to consider how to break it into
+smaller chunks before proceeding.
+
+pvlib-python contains :ref:`3 "layers" of code <modeling-paradigms>`:
+functions, PVSystem/Location, and ModelChain. We recommend that
+contributors focus their work on only one or two of those layers in a
+single pull request. New models are *not* required to be available to
+the higher-level API!
```

Comment:
It's hard to see a ton of value in adding models that aren't accessible to (perhaps many?) users through the higher-level API. Perhaps touch on this point here: What is the value and/or subsequent steps to make a new model "fully" accessible?

Reply:
I'm OK with what Will wrote here. At the last workshop we heard from a broad sample of users that pvlib (either MATLAB or python) serves two primary roles: as a reference library of algorithms (our original intent), and as a platform for research or custom modeling to supplement commercial applications. `ModelChain` is valuable, although not essential, to these use cases. So it's OK, from my point of view, to add model functions and expose them at the `PVSystem/Location` level but not the `ModelChain` level.

Reply:
I think Cliff said it well. I'd add that I'm not sure that ``ModelChain`` is going to exist in anything like its current form in 3-5 years, but I am sure that most of the core functions will remain fairly similar to what they are now.